src/plugins/dashboard/public/dashboard_container/state/diffing/dashboard_diffing_integration.ts

```diff
@@ -99,10 +98,14 @@ export function startDiffingDashboardState(
               .bind(this)(lastSavedInput, currentInput)
               .then((unsavedChanges) => {
                 if (observer.closed) return;
+                const validUnsavedChanges = omit(unsavedChanges, keysNotConsideredUnsavedChanges);
+                const hasChanges = Object.keys(validUnsavedChanges).length > 0;
+                this.hasUnsavedChanges.next(hasChanges);
 
-                updateUnsavedChangesState.bind(this)(unsavedChanges);
                 if (creationOptions?.useSessionStorageIntegration) {
-                  backupUnsavedChanges.bind(this)(unsavedChanges);
+                  this.backupUnsavedChanges.next(
+                    omit(unsavedChanges, keysToOmitFromSessionStorage)
+                  );
```

Comment:
IIUC, this method pushes updates to the `hasUnsavedChanges` and `backupUnsavedChanges` BehaviorSubjects. The dispatch that calls the `setHasUnsavedChanges` reducer is now handled by the subscriptions to those BehaviorSubjects. If that is correct, should we update the comment on lines 78-80?

Reply:
Ah, good catch! Done in bd48fc9e2fe01e134a8b5c4efd22ec0bdb0c0c64